jte-spring-boot-starter-2/src/main/java/gg/jte/springframework/boot/autoconfigure/JteView.java

```diff
@@ -24,16 +24,10 @@ public boolean checkResource(Locale locale) {
 
     @Override
     protected void renderMergedTemplateModel(Map<String, Object> model, javax.servlet.http.HttpServletRequest request, javax.servlet.http.HttpServletResponse response) throws Exception {
-
         String url = this.getUrl();
-
-        Utf8ByteOutput output = new Utf8ByteOutput();
-        templateEngine.render(url, model, output);
-
         response.setContentType(MediaType.TEXT_HTML_VALUE);
         response.setCharacterEncoding(StandardCharsets.UTF_8.name());
-        response.setContentLength(output.getContentLength());
-
-        output.writeTo(response.getOutputStream());
+        PrintWriterOutput output = new PrintWriterOutput(response.getWriter());
```

Comment:
Is there a reason for changing this?

Because if we do this, the `templateEngine.setBinaryStaticContent(true);` needs to be removed, otherwise during rendering all pre-encoded byte arrays will be converted to Strings during rendering.

I don't know how Springs `PrintWriter` behaves. But usually a `PrintWriter` flushes its content from time to time. Have you tested what happens if there is a exception during rendering? Does the user then get a half rendered page?

Reply:
Ah, I've seen your commit message with the reason behind this change.

`templateEngine.setBinaryStaticContent(true);` should then still be removed, in order to not make the rendering unreasonably slow.

Reply:
Done :)